From 66552d8871e39a3e2f24bf889f175d5c26aca3d6 Mon Sep 17 00:00:00 2001
From: Jo Shields <jo.shields@xamarin.com>
Date: Wed, 2 Dec 2015 14:36:37 +0000
Subject: [PATCH] Assume we are inside XBuild when 4.5/Mono.Posix.dll exists

4.0/Mono.Posix.dll is not guaranteed to exist, on Mono 4.0+
---
 lib/bootstrap/4.0/Microsoft.FSharp.Targets       | 2 +-
 src/fsharp/FSharp.Build/Microsoft.FSharp.Targets | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/bootstrap/4.0/Microsoft.FSharp.Targets b/lib/bootstrap/4.0/Microsoft.FSharp.Targets
index b8c51d4..5d6fe98 100644
--- a/lib/bootstrap/4.0/Microsoft.FSharp.Targets
+++ b/lib/bootstrap/4.0/Microsoft.FSharp.Targets
@@ -68,7 +68,7 @@ Copyright (C) Microsoft Corporation. Apache 2.0 License.
 
   <PropertyGroup>
     <UsingXBuild>false</UsingXBuild>
-    <UsingXBuild Condition="Exists('$(MSBuildExtensionsPath32)\..\4.0\Mono.Posix.dll')">true</UsingXBuild>
+    <UsingXBuild Condition="Exists('$(MSBuildExtensionsPath32)\..\4.5\Mono.Posix.dll')">true</UsingXBuild>
   </PropertyGroup>
 
   
diff --git a/src/fsharp/FSharp.Build/Microsoft.FSharp.Targets b/src/fsharp/FSharp.Build/Microsoft.FSharp.Targets
index 5ff6508..66f521c 100644
--- a/src/fsharp/FSharp.Build/Microsoft.FSharp.Targets
+++ b/src/fsharp/FSharp.Build/Microsoft.FSharp.Targets
@@ -68,7 +68,7 @@ Copyright (C) Microsoft Corporation. Apache 2.0 License.
 
   <PropertyGroup>
     <UsingXBuild>false</UsingXBuild>
-    <UsingXBuild Condition="Exists('$(MSBuildExtensionsPath32)\..\4.0\Mono.Posix.dll')">true</UsingXBuild>
+    <UsingXBuild Condition="Exists('$(MSBuildExtensionsPath32)\..\4.5\Mono.Posix.dll')">true</UsingXBuild>
   </PropertyGroup>
 
     <Target
-- 
2.5.0

